### DISCUSSION TITLE
Recherche on suitable packages for Equi7Grid v2.0

### DISCUSSION THREAD

**Discussion by bbauerma** [TUW-GEO/Equi7Grid#44] [Ideas]
### Let's collect candidate packages for the grid core package. We seek for packages that are ...
- living 
- supporting common standards
- solving the already reported [issues](/TUW-GEO/Equi7Grid/issues)
- helping us to enable functions gathered in #45

**Comment by sebhahn** [comment]
- https://github.com/cogeotiff/rio-tiler-mosaic
- https://github.com/developmentseed/mosaicjson-spec
- https://geoparquet.org/
- https://github.com/mapbox/mercantile

**Reply by bbauerma** [reply]
mercantile: databases, etc...
geoparquet: dataformat for vectordata, geojson
rio-tiler: mosaic from a bunch files which can overlap

**Comment by bbauerma** [comment]
as on outlook, here are OGC APIs:
https://ogcapi.ogc.org/
and the OGC nomenclature for DGGS, tiles, zones, and more:
https://docs.ogc.org/as/20-040r3/20-040r3.pdf

**Comment by bbauerma** [comment]
some more ideas from the initial meeting on 18.6.2024:
https://developmentseed.org/titiler/
Equi7 clone for NASA's equal area solution: https://github.com/measures-glance/glance-grids
very interesting, but archived in 2020: https://github.com/ceholden/stems
and of course https://github.com/TUW-GEO/geospade

**Reply by bbauerma** [reply]
titiler: tile server, etc; used by EODC for sigma0 datacube for visualisation

**Comment by sebhahn** [comment]
this seems to be a good candidate: https://databrickslabs.github.io/mosaic/index.html

**Reply by sebhahn** [reply]
seems to depend on pyspark

**Comment by bbauerma** [comment]
A very interesting recent development, kudos to [@csaybar](https://github.com/csaybar)
https://github.com/IPL-UV/equi7grid_lite

**Comment by cnavacch-spire** [comment]
- https://github.com/digitalglobe/tiletanic 
    - not useful, more tailored towards WMTS 
    - not active anymore
- https://github.com/tilezen/mapbox-vector-tile
    - not useful, more tailored towards WMTS
    - limited functionality
    - active
    - big user community and many contributors
- https://github.com/developmentseed/morecantile
    - active
    - light-weighted
    - many contributors
    - follows OGC standards
    - very similar to geospade, but again tailored towards WMTS (zooming levels, quadtree etc.)

**Comment by sebhahn** [comment]
spatial indexing
- https://rtree.readthedocs.io/
- https://github.com/karimbahgat/Pyqtree
- https://geopandas.org/en/stable/docs/reference/sindex.html

**Reply by csaybar** [reply]
Maybe in an early release, but zero-copy methods for this look promising! 
https://github.com/kylebarron/geo-index

**Comment by bbauerma** [comment]
we were just made aware of this new feature of GDAL, from a quick glance perfectly matching our use case: 
[GDAL Raster Tile Index](https://gdal.org/drivers/raster/gti.html)

**Reply by bbauerma** [reply]
file based operations.
useful for reprojection tasks

**Comment by sebhahn** [comment]
tilematrix - https://www.ogc.org/standard/tms/
https://github.com/ungarj/tilematrix

**Reply by bbauerma** [reply]
realisation of the OCG terminology, slim and matching our case?
well documented and tested
developed by single guy in Vienna

**Comment by bbauerma** [comment]
the outcome of our latest meeting was:
let's us tryout these two packages, and start two git branches trying if they serve the collected needs:
1) https://github.com/ungarj/tilematrix
2) https://github.com/IPL-UV/equi7grid_lite
